nixos/modules/tasks/filesystems.nix

```diff
@@ -107,6 +107,17 @@ let
         '';
       };
 
+      autoExpand = mkOption {
```

Comment:
This looks like a new feature. It would be easier to review this PR and move it forward if you isolated this feature and opened a separate PR for it.